smartdispatch/tests/test_smartdispatch_script.py

```
@@ -0,0 +1,87 @@
+import unittest
+from smartdispatch import smartdispatch_script
+import subprocess
+from mock import patch
+import tempfile as tmp
+import shutil
+import six
+import sys
+
+class TestSmartScript(unittest.TestCase):
+
+    def setUp(self):
+        self._base_dir = tmp.mkdtemp()
+        smartdispatch_script.LOGS_FOLDERNAME = self._base_dir
+
+    def tearDown(self):
+        shutil.rmtree(self._base_dir)
+
+    def test_gpu_check(self):
+
+        argv = ['-x', '-g', '2', '-G', '1', '-C', '1', '-q', 'random', '-t', '00:00:10' ,'launch', 'echo', 'testing123']
+
+        # Test if the check fail
+        with self.assertRaises(SystemExit) as context:
+            smartdispatch_script.main(argv=argv)
+
+        self.assertTrue(context.exception.code, 2)
+
+        # Test if the test pass
+        argv[2] = '0'
+
+        try:
+            smartdispatch_script.main(argv=argv)
+        except SystemExit as e:
+            self.fail("The command failed the check, but it was supposed to pass.")
```

What happens if you just let it crash on smartdispatch_script.main()? Is it bad because of SystemExit? Do you lose the stacktrace or a valuable error message by doing self.fail?